### PR TITLE
Remove/rename some functionality and move operators into submodule

### PR DESCRIPTION
**Pull request by oschulz**
@cscherrer these is the first PR to implement the somewhat breaking changes we discussed over the last days (PRs to be collected on the mt-015 branch for a MeasureBase v0.15 release sometime next week).

Most of the removed/renamed functionality is rarely used, and the operators can be accessed via `using MeasureBase.MeasureOperators` now, so user code should be easy to adapt.

PR has a clean history, we should **not** squash it when we merge into mt-015, so we can keep track more easily.

### CODE REVIEW

File: src/combinators/bind.jl [hunk 1]
```diff
@@ -3,31 +3,30 @@ struct Bind{M,K} <: AbstractMeasure
...
-probabilistic programming language like Soss.jl, this could be expressed as
+- a measure μ
+- a kernel function k that takes values from the support of μ and returns a
+  measure
 
-Note that bind is usually written `>>=`, but this symbol is unavailable in Julia.
+The *monadic bind* operation `mbind(k, μ)` returns is a new measure.
+
+A monadic bind ofen written as `>>=` (e.g. in Haskell), but this symbol is
+unavailable in Julia.
```
Comment (cscherrer) [inline src/combinators/bind.jl:19]:
```suggestion
The *monadic bind* operation `mbind(k, μ)` returns is a new measure.
If `ν == mbind(k, μ)` and all measures involved are sampleable, then 
samples from `rand(ν)` follow the same distribution as those from `rand(k(rand(μ)))`.


A monadic bind ofen written as `>>=` (e.g. in Haskell), but this symbol is
unavailable in Julia.
```

File: src/combinators/bind.jl [hunk 2]
```diff
@@ -3,31 +3,30 @@ struct Bind{M,K} <: AbstractMeasure
...
+    pushfwd(Base.Fix1(*, scale), StdNormal())
 end
 ```
-
-See also `bind` and `Bind`
 """
-↣(μ, k) = bind(μ, k)
-
-bind(μ, k) = Bind(μ, k)
+mbind(k, μ) = Bind(μ, k)
```
Comment (cscherrer) [inline src/combinators/bind.jl:28]: Should we also swap the arguments in `Bind`?

File: src/combinators/likelihood.jl [hunk 3]
```diff
@@ -91,34 +91,35 @@ Similarly to the above, we have
...
 
-``P(θ|x) ∝ P(θ) P(x|θ)``
+``P(θ|x) ∝ P(x|θ) P(θ)``
 
-The product on the right side is computed pointwise. To work with this in
-MeasureBase, we have a "pointwise product" `⊙`, which takes a measure and a
-likelihood, and returns a new measure, that is, the unnormalized posterior that
-has density ``P(θ) P(x|θ)`` with respect to the base measure of the prior.
+In measure theory, the product on the right side is actually the Lebesgue integral,
+of the likelihood with respect to the prior.
```
Comment (cscherrer) [inline src/combinators/likelihood.jl:97]:
```suggestion
In measure theory, the product on the right side is the Lebesgue integral
of the likelihood with respect to the prior.
```

File: src/combinators/likelihood.jl [hunk 4]
```diff
@@ -91,34 +91,35 @@ Similarly to the above, we have
...
 struct Likelihood{K,X} <: AbstractLikelihood
     k::K
     x::X
 
-    Likelihood(k::K, x::X) where {K<:AbstractTransitionKernel,X} = new{K,X}(k, x)
-    Likelihood(k::K, x::X) where {K<:Function,X} = new{K,X}(k, x)
-    Likelihood(μ, x) = Likelihood(kernel(μ), x)
+    Likelihood(k::K, x::X) where {K,X} = new{K,X}(k, x)
+#!!!!!!!!!!!    # For type stability if `K isa UnionAll (e.g. a parameterized MeasureType)`
+    Likelihood(::Type{K}, x::X) where {K<:AbstractMeasure,X} = new{K,X}(K, x)
```
Comment (cscherrer) [inline src/combinators/likelihood.jl:122]: @cscherrer review this

File: src/measure_operators.jl [hunk 5]
```diff
@@ -0,0 +1,141 @@
+"""
+    module MeasureOperators
+
+Defines the following operators for measures:
+
+* `f ⋄ μ == pushfwd(f, μ)`
+
+* `μ ∘ f == inverse(f) ⋄ μ`
```
Comment (cscherrer) [inline src/measure_operators.jl:8]:
```suggestion
* `μ ⊙ f == inverse(f) ⋄ μ`
```

File: src/combinators/bind.jl (hunk 2 above)
Comment (oschulz) [reply src/combinators/bind.jl]: Done.

**Review by cscherrer — DISMISSED** [review]

**Pull request merged**